librad/src/keys/device.rs

```diff
@@ -137,6 +139,24 @@ impl Signature {
     pub fn verify(&self, data: &[u8], pk: &PublicKey) -> bool {
         sign::verify_detached(self, &data, pk)
     }
+
+    pub fn from_hex_string(s: &str) -> Result<Self, failure::Error> {
+        let bytes =
+            decode(s).map_err(|_| failure::format_err!("Cannot decode signature hex string"))?;
+        let buffer = if bytes.len() == 64 {
+            let mut buffer = [0u8; 64];
+            for (i, v) in bytes.iter().enumerate() {
```

Comment:
I think you could us [copy_from_slice](https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.copy_from_slice) i.e. `buffer.copy_from_slice(&bytes)`